homeassistant/components/axis/config_flow.py

```diff
@@ -147,40 +119,39 @@ def __init__(self):
         title = f"{self.model} - {self.serial_number}"
         return self.async_create_entry(title=title, data=data)
 
-    async def _update_entry(self, entry, host):
-        """Update existing entry if it is the same device."""
+    async def _update_entry(self, entry, host, port):
+        """Update existing entry."""
+        if (
+            entry.data[CONF_DEVICE][CONF_HOST] == host
+            and entry.data[CONF_DEVICE][CONF_PORT] == port
+        ):
+            return self.async_abort(reason="already_configured")
+
         entry.data[CONF_DEVICE][CONF_HOST] = host
+        entry.data[CONF_DEVICE][CONF_PORT] = port
+
         self.hass.config_entries.async_update_entry(entry)
+        return self.async_abort(reason="updated_configuration")
 
     async def async_step_zeroconf(self, discovery_info):
-        """Prepare configuration for a discovered Axis device.
+        """Prepare configuration for a discovered Axis device."""
+        serial_number = discovery_info["properties"]["macaddress"]
 
-        This flow is triggered by the discovery component.
-        """
-        serialnumber = discovery_info["properties"]["macaddress"]
-
-        if serialnumber[:6] not in AXIS_OUI:
+        if serial_number[:6] not in AXIS_OUI:
             return self.async_abort(reason="not_axis_device")
 
         if discovery_info[CONF_HOST].startswith("169.254"):
             return self.async_abort(reason="link_local_address")
 
-        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
-        self.context["macaddress"] = serialnumber
-
-        if any(
-            serialnumber == flow["context"]["macaddress"]
-            for flow in self._async_in_progress()
-        ):
-            return self.async_abort(reason="already_in_progress")
-
-        device_entries = configured_devices(self.hass)
-
-        if serialnumber in device_entries:
-            entry = device_entries[serialnumber]
-            await self._update_entry(entry, discovery_info[CONF_HOST])
-            return self.async_abort(reason="already_configured")
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if serial_number == entry.unique_id:
+                return await self._update_entry(
```

Comment:
:+1 Yeah, I also plan to allow updating from manually initiating the config flow, in case discovery is turned off or doesn't work or is not supported.